cldfbench_nominalperson_cldf.py

```diff
@@ -20,63 +20,69 @@ def cmd_download(self, args):
 
         >>> self.raw_dir.download(url, fname)
         """
-        pass
+        wals_dir = self.raw_dir / 'wals'
+        if not wals_dir.exists():
+            wals_dir.mkdir()
+        rec = Record.from_doi('10.5281/zenodo.13950591')
+        rec.download_dataset(wals_dir)
 
     def cmd_makecldf(self, args):
         # Extend the schema:
-        args.writer.cldf.add_component('LanguageTable')
-        args.writer.cldf.add_columns('LanguageTable','Walscode')    # add non-standard column
-        
-        args.writer.cldf.add_component('CodeTable')
+        args.writer.cldf.add_component(
+            'LanguageTable',
+            'Walscode',
+            {'name': 'ISOcodes', 'separator': ';', 'datatype': {'base': 'string', 'format': '[a-z]{3}'}})
+
+        #args.writer.cldf.add_component('CodeTable')
         args.writer.cldf.add_component('ParameterTable')
         args.writer.cldf.add_component('ValueTable')
         args.writer.cldf.add_component('ExampleTable')
 
+        glangs = {lg.id: lg for lg in args.glottolog.api.languoids()}
+
+        wals_codes = {r['ID']: r['Name'] for r in self.raw_dir.joinpath('wals').read_csv('codes.csv', dicts=True)}
+        wals_83A = {
+            r['Language_ID']: wals_codes[r['Code_ID']]
+            for r in self.raw_dir.joinpath('wals').read_csv('values.csv', dicts=True)
+            if r['Parameter_ID'] == '83A'}
+
+        params = ['apc_order', 'constituent_order']
+        for param in ['apc_order', 'constituent_order']:
+            args.writer.objects['ParameterTable'].append(dict(
+                ID=param,
+                Name=param.replace('_', ' '),
+            ))
+
         # Load data from grammarchecks
-        
-        lang_id=0
+        lids = set()
         val_id=0
         for row in self.raw_dir.read_csv('grammarchecks.csv',dicts=True):
+            if row['constituent_order_wals']:
+                assert row['constituent_order_wals'] == wals_83A[row['walscode']], '{}: {} vs. {}'.format(row['walscode'], row['constituent_order_wals'], wals_83A[row['walscode']])
             if row['include']=='y':
+                assert row['glottocode'] in glangs
+                lids.add(row['glottocode'])
```

Comment:
Instead of trying to read partially written CLDF data (which won't work), we can just construct lookup tables in the code.